source/wp-content/themes/wporg-developer-2023/src/command-content/block.php

```diff
@@ -0,0 +1,108 @@
+<?php
+namespace WordPressdotorg\Theme\Developer_2023\Dynamic_Command_Content;
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+
+/**
+ * Registers the block using the metadata loaded from the `block.json` file.
+ * Behind the scenes, it registers also all assets so they can be enqueued
+ * through the block editor in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/reference/functions/register_block_type/
+ */
+function init() {
+	register_block_type(
+		dirname( dirname( __DIR__ ) ) . '/build/command-content',
+		array(
+			'render_callback' => __NAMESPACE__ . '\render',
+		)
+	);
+}
+
+/**
+ * Render the block content.
+ *
+ * @return string Returns the block markup.
+ */
+function render( $attributes, $content, $block ) {
+
+	$content = filter_the_content( get_the_content() );
```

Comment:
`get_the_content` doesn't run the `the_content` filter, so I think `filter_the_content` is unnecessary here— since it's meant to undo some of the special character magic WordPress does on `the_content`. I'm not 100% sure about that, though.

Reply:
> get_the_content doesn't run the the_content filter, so I think filter_the_content is unnecessary here— since it's meant to undo some of the special character magic WordPress does on the_content. I'm not 100% sure about that, though.

TIL. I'll remove it first, I don't see anything go weird after removing it.

Reply:
Possibly a really basic question. For not calling global calls here, is it because blocks sometimes would be used in the env that has no global variables? Is there any example of it?

Reply:
Yeah, the blocks could be used in a Query Loop on another page — so the global functions would refer to the page itself, not the current item in the query loop. GB has some protections in place to avoid that, but I've run into issues related to that (like [this one](https://github.com/WordPress/pattern-directory/pull/553)).

We're not doing anything like that on this project at the moment, but I see it as future-proofing things.

Reply:
Thanks for the clarification!